src/lib/yang/yang.lua

```diff
@@ -14,16 +15,36 @@ load_schema_by_name = schema.load_schema_by_name
 add_schema = schema.add_schema
 add_schema_file = schema.add_schema_file
 
-load_config_for_schema = data.load_config_for_schema
-load_config_for_schema_by_name = data.load_config_for_schema_by_name
+function load_config_for_schema (schema, stream)
+   local data = data.load_config_for_schema(schema, stream)
+   path_data.consistency_checker_from_schema(schema, true)(data)
+   return data
+end
+function load_config_for_schema_by_name(schema_name, stream)
+   local data = data.load_config_for_schema_by_name(schema_name, stream)
+   path_data.consistency_checker_from_schema_by_name(schema_name, true)(data)
+   return data
+end
 
 print_config_for_schema = data.print_config_for_schema
 print_config_for_schema_by_name = data.print_config_for_schema_by_name
 
-compile_config_for_schema = binary.compile_config_for_schema
-compile_config_for_schema_by_name = binary.compile_config_for_schema_by_name
+function compile_config_for_schema(schema, data, filename, source_mtime)
+   local data = binary.compile_config_for_schema(schema, data, filename, source_mtime)
```

Comment:
I think we can avoid the check here, and also below when loading compiled configs.  Let's assume that we run the check when loading from source.  Then at what point can consistency be broken?  Answer: only if the configuration is modified in an ad-hoc way, programmatically.  It's *then* that we need to run the checks -- after a modification.  Unfortunately there's not much we can do at that point except error out; we have no capacity for rollback :(  Anyway there's no need to do the check when compiling data, as the data should be consistent already, nor when loading compiled data, as the data should have been consistent before it was compiled!